packages/bot/src/scratch/hooks/toolbox.js

```diff
@@ -198,18 +198,24 @@ Blockly.Toolbox.prototype.showCategory_ = function (category_id) {
         flyout_content.blocks = flyout_content.blocks.concat(uniqueProceBlocks);
     } else {
         const selected_category = this.categoryMenu_.categories_.find(category => category.id_ === category_id);
-        flyout_content = selected_category.getContents();
-
-        // Dynamic categories
-        if (typeof flyout_content === 'string') {
-            const fnToApply = this.workspace_.getToolboxCategoryCallback(flyout_content);
-            flyout_content = fnToApply(this.workspace_);
-        }
+        flyout_content = this.getCategoryContents(selected_category, this.workspace_);
     }
 
     flyout.setContents(flyout_content);
 };
 
+Blockly.Toolbox.prototype.getCategoryContents = function(selected_category, workspace) {
```

Comment:
No need to have a `workspace` instance passed, since the function is part of the prototype we may reference it with `this.workspace_`.